### PR TITLE
Implement `cargo lint` and fix some clippy errors

### PR DESCRIPTION
**Pull request by alanhdu**
This creates a `cargo lint` command that runs clippy with certain lints disabled. I've also gone ahead and fixed some of the lint errors, although there are many more still to go.

cc #848

### CODE REVIEW

File: crates/ra_assists/src/auto_import.rs [hunk 1]
```diff
@@ -334,7 +334,7 @@ fn best_action_for_target<'b, 'a: 'b>(
         .filter_map(ast::UseItem::use_tree)
         .map(|u| walk_use_tree_for_best_action(&mut storage, None, u, target))
         .fold(None, |best, a| {
-            best.and_then(|best| Some(*ImportAction::better(&best, &a))).or(Some(a))
+            best.and_then(|best| Some(*ImportAction::better(&best, &a))).or_else(|| Some(a))
```
Comment (matklad) [inline crates/ra_assists/src/auto_import.rs]: I feel conflicted about this, it seems like closure creates noise for little benefit here. But I guess it's ok?

File: crates/ra_project_model/src/lib.rs [hunk 2]
```diff
@@ -183,7 +183,7 @@ impl ProjectWorkspace {
                         if let (Some(&from), Some(&to)) =
                             (sysroot_crates.get(&from), sysroot_crates.get(&to))
                         {
-                            if let Err(_) = crate_graph.add_dep(from, name.into(), to) {
```
Comment (matklad) [inline crates/ra_project_model/src/lib.rs]: Let's disable this lint? I find `if let` somewhat more readable, and, with result, `Err(_)` also sends a strong signal that an error is ignored.

File: crates/ra_assists/src/auto_import.rs (hunk 1 above)
Comment (alanhdu) [reply crates/ra_assists/src/auto_import.rs]: Yeah, I think the fact that clippy interprets these constructors as "function calls" is pretty unfortunate, It looks like a bug (https://github.com/rust-lang/rust-clippy/issues/1609), but for now I figured that it'd be better to make the changes just so we can get all the lint errors out of the way.

**Pull request merged**